### ISSUE TITLE
Unexpected segfaults when solving problems via the mps/dec interface

### ISSUE THREAD

**Issue by MathiasPBerger** [Argonne-National-Laboratory/DSP#206]
Dear DSP developers,

I have been trying to solve some problems with DSP using the mps/dec interface and have encountered some issues. Please see below for details:

I built the DSP executable and library on three separate machines and operating systems, namely:

1)	on OS X 10.15.7 with CPLEX 12.10 and Gurobi 9.0.3, after git cloning from master (commit  9706f7f)
2)	on CentOS Linux 7 with CPLEX 12.9 and Gurobi 9.1.1, after git cloning from master (commit 6675d30)
3)	on Ubuntu 18.04 installed via the Windows Subsystem for Linux, with CPLEX 20.1 and Gurobi 9.1.3, after git cloning from master (commit 6675d30). 

The implementation was tested with ctest on CentOS, and with the noswot.mps/dec example on all machines, which worked fine.

Next, two mps files representing the same model were generated. The first one was generated via Clp’s Python interface (CyLP), while the second one was generated with CPLEX's Python interface. The files were tested with CPLEX and Gurobi via their command line interfaces, and both solvers returned an optimal solution with objective 146.6 on each machine.

The DSP implementations were then tested with the aforementioned mps files and an associated dec file. More specifically, the ExtendedForm and DW algorithms were tested. In both cases, it seems that DSP could successfully read the files and load the model. However, both algorithms segfaulted on each machine when trying to solve the model. The mps/dec files as well as the logs produced on the CentOS machine can be found in the [files.zip archive](https://github.com/Argonne-National-Laboratory/DSP/files/7719896/files.zip).

It is not entirely clear what is going, and I was therefore wondering whether you could provide some input on this?

I look forward to hearing from you,

Best regards,

Mathias Berger

**Comment by kibaekkim** [comment]
@MathiasPBerger Thanks for the report. We will take a look into that and get back to you.

**Comment by hideakiv** [comment]
Hi, I was able to replicate this issue on macOS 11.6.1 Gurobi 9.1.2. Let me get back to you once we figure out what is going on.

**Comment by MathiasPBerger** [comment]
Great, thank you for taking the time to look into it. Please let me know if I can assist you in any way.

**Comment by hideakiv** [comment]
@MathiasPBerger The issue seems to be caused by .dec file containing "\r\n" at the end of each line. I added some code to remove "\r" at the end. Would you be able to check if this works for you?

**Comment by MathiasPBerger** [comment]
@hideakiv I built a new version of DSP from branch issue/206. This partially fixed the problem, in that the code does not segfault anymore. However, the solutions returned by DE and DW do not seem to make much sense. More specifically, DE finishes with status 3002 (which is not optimal, as far as I understand), while DW finishes with status 3000 but returns what looks like a trivial solution. Please see the associated [DE](https://github.com/Argonne-National-Laboratory/DSP/files/7731003/dsp_de.log) and [DW](https://github.com/Argonne-National-Laboratory/DSP/files/7731005/dsp_dw.log) logs.

**Comment by hideakiv** [comment]
@MathiasPBerger Thanks for the feedback. Let me take a look...

**Comment by hideakiv** [comment]
@MathiasPBerger In your microgrid_1d_clp.dec file, you have R_1_344 that doesn't exist in the mps file and you are missing R_1_0. By any chance, is the number offset by 1?

**Comment by MathiasPBerger** [comment]
@hideakiv Thank you for pointing this out, there was indeed an offset issue with the dec file. This has now been corrected (see [this file](https://github.com/Argonne-National-Laboratory/DSP/files/7734438/microgrid_1d.dec.zip)). The DE algorithm runs fine and returns the same objective as Gurobi and CPLEX, which is great. The solution returned by the DW algorithm still looks weird, however. Please see the associated [DE](https://github.com/Argonne-National-Laboratory/DSP/files/7734470/dsp_de.log) and [DW](https://github.com/Argonne-National-Laboratory/DSP/files/7734477/dsp_dw.log) logs. I'm gonna do some digging on my end and check the problem formulation.

**Comment by hideakiv** [comment]
@MathiasPBerger There are several rows that exist both in block 4 and Masterconss. Is this intended?

**Comment by MathiasPBerger** [comment]
@hideakiv The repeated rows were not supposed to be there, thank you for pointing it out. I checked the problem formulation and ran a couple of tests on my end. Results suggest that the problem does not come from the mps/dec interface of DSP. More specifically, I took the same formulation and bounded all variables to avoid unbounded subproblems. The objective of the solution returned by the Dantzig-Wolfe algorithm now matches the one returned by the extensive form solve. So I guess the problem rather comes from the fact that in its present form, the DW algorithm does not handle unbounded subproblems, as suggested in issue 211. I'll therefore close this issue. Thank you for being so responsive and helpful!
